ui/src/app/pages/app/component.ts

```diff
@@ -93,14 +102,95 @@ export class AppComponent implements OnInit {
         queryParamsStore.next(queryParams);
         routeConfigDataStore.next(routeConfig.data);
       }
-    });
+    }));
 
-    routeDataStore.subscribe(({title, pathElementForTitle}) => {
+    this.subscriptions.push(routeDataStore.subscribe(({title, pathElementForTitle}) => {
       this.zone.run(() => {
         this.setTitleFromReactRoute({title, pathElementForTitle});
         this.initialSpinner = false;
       });
-    });
+    }));
+
+
+    // TODO angular2react: this active icon stuff can move into help-sidebar once it reaches a state where it
+    // doesn't remount on every navigation event
+    setSidebarActiveIconStore.next(localStorage.getItem(LOCAL_STORAGE_KEY_SIDEBAR_STATE));
+
+    this.subscriptions.push(routeDataStore.subscribe((newRoute, oldRoute) => {
+      if (!fp.isEmpty(oldRoute) && !fp.isEqual(newRoute, oldRoute)) {
+        setSidebarActiveIconStore.next(null);
+      }
+    }));
+
+    this.subscriptions.push(urlParamsStore
+      .map(({ns, wsid}) => ({ns, wsid}))
+      .debounceTime(1000) // Kind of hacky but this prevents multiple update requests going out simultaneously
+      // due to urlParamsStore being updated multiple times while rendering a route.
+      // What we really want to subscribe to here is an event that triggers on navigation start or end
+      // Debounce 1000 (ms) will throttle the output events to once a second which should be OK for real life usage
```

Comment:
Famous last words..

Probably an important thing to note here is that the `recent resources` stuff is pretty much a best effort. Debouncing this by every second is totally fine, since we don't mind missing an occasional event.

Reply:
yeah.. I would have been more careful for something more important but this seemed fine here